sdk/helper/automatedrotationutil/fields.go

```diff
@@ -104,3 +114,77 @@ func AddAutomatedRotationFields(m map[string]*framework.FieldSchema) {
 		m[name] = schema
 	}
 }
+
+// HandleRotationJobOperation is a helper method wrapper to the two individual helper methods
+// that register and deregister a rotation job each, and returns a standardized error response
+// with a warning in the event of an unexpected error.
+// revertStorageFunc is a function that should revert the original config back in storege before
+// any fields were updated.
+func (p *AutomatedRotationParams) HandleRotationJobOperation(ctx context.Context, b *framework.Backend, req *logical.Request, revertStorageFunc func() error) *logical.Response {
+	if p.DisableAutomatedRotation {
+		// Disable automated rotation and deregister credentials if required.
+		if err := p.HandleDeregisterRotationJob(ctx, b, req); err != nil {
+			resp := logical.ErrorResponse("error deregistering rotation job but config was successfully updated: %s", err)
+			resp.AddWarning("config was successfully updated despite failing to disable automated rotation")
+
+			if err := revertStorageFunc(); err != nil {
```

Comment:
not a blocker for the current implementations in plugins: Should we allow `revertStorageFunc` to be `nil`? Wondering if we should let plugins control whether they want to set that. If we add in a nil check for it before calling it, that would make it so plugins can also pass in `nil` for cases where they do not want to revert storage writes

Reply:
What do we do with the error we receive from the storage revert?